client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs

```diff
@@ -69,11 +69,14 @@ public static class AutoRegistrationInMonitoring
                     if (monitoringRegistration.Url == myMonitoringUrl)
                         return;
 
-                    log.WriteMonitor("Auto-registration in monitoring", podTag, $"There is a registration for {myMonitoringName} in monitoring service!");
+                    if (monitoringRegistration.Url != _missingEnvVarUrl)
```

Comment:
Seems like this If is guaranteed to be always true when we do not have a monitoring url provided in settings 'cause of the line 69. Also, this code is unreachable in that case.
Btw, it's not so clear (for me personally) why does the `monitoringService.GetService(...)` return "0.0.0.0" as the url when monitoring is already registered.

Reply:
Won't it be more clear with your intension if `GetService(...)` will return a tuple, where the first part will contain some `Enum` with monitoring registration (current) state, and the second part will be `monitoringRegistration` info?

Reply:
"Seems like this If is guaranteed to be always true" - not true,
Line 69 prevents service from registration with the same data.
Line 72 handles the case when initiall there was no MyMonitoringUrl env var defined and this app was registered with _missingEnvVarUrl. So when this env var was added, proper url can replace  _missingEnvVarUrl. Without it not only original _missingEnvVarUrl will stay, additional registration with _missingEnvVarUrl will be added by Line 78

Reply:
About suggestion of GetService result change - we can do it, but in another task.